rust/main/lander/src/dispatcher/stages/inclusion_stage.rs

```diff
@@ -111,9 +111,17 @@ impl InclusionStage {
         state: DispatcherState,
         domain: String,
     ) -> Result<(), LanderError> {
+        let base_interval = *state.adapter.estimated_block_time();
+
         loop {
-            // evaluate the pool every block
-            sleep(Duration::from_millis(10)).await;
+            // Use adaptive polling interval based on block time, but never faster than 100ms
+            // for small block time chains, and never slower than 1/4 block time for responsiveness
+            let polling_interval = std::cmp::max(
+                base_interval / 4,          // Never slower than 1/4 block time for responsiveness
+                Duration::from_millis(100), // Never faster than 100ms to avoid excessive RPC calls
+            );
```

Comment:
Sirce `base_interval` does not change, the whole calculation can be done outside of the loop.